// @koyeb/api-client-js/dist/sdk.gen.js
// This file is auto-generated by @hey-api/openapi-ts
import { client } from './client.gen.js';
/**
 * List Branches
 */
export const listBranches = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/git/branches',
    ...options
});
/**
 * List Repositories
 */
export const listRepositories = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/git/repositories',
    ...options
});
/**
 * Resync Organization
 */
export const resyncOrganization = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/git/sync/organization/{organization_id}',
    ...options
});
/**
 * Get organization usage
 */
export const getOrganizationUsage = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/usages',
    ...options
});
/**
 * Get organization usage details
 */
export const getOrganizationUsageDetails = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/usages/details',
    ...options
});
/**
 * List Regional Deployment events
 */
export const listRegionalDeploymentEvents = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/regional_deployment_events',
    ...options
});
/**
 * List Regional Deployments
 *
 * Experimental: use at your own risk
 */
export const listRegionalDeployments = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/regional_deployments',
    ...options
});
/**
 * Get Regional Deployment
 *
 * Experimental: use at your own risk
 */
export const getRegionalDeployment = (options) => (options.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/regional_deployments/{id}',
    ...options
});
/**
 * Create resources from compose.
 */
export const compose = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/compose',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Verify Docker Image
 *
 * Verify if a docker image is reachable
 */
export const verifyDockerImage = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/docker-helper/verify',
    ...options
});
/**
 * Get Deployment Scaling
 */
export const getDeploymentScaling = (options) => (options.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/deployment/{id}/scaling',
    ...options
});
/**
 * List Deployment events
 */
export const listDeploymentEvents = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/deployment_events',
    ...options
});
/**
 * List Deployments
 */
export const listDeployments = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/deployments',
    ...options
});
/**
 * Get Deployment
 */
export const getDeployment = (options) => (options.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/deployments/{id}',
    ...options
});
/**
 * Cancel Deployment
 *
 * Deployment cancellation is allowed for the following status:
 * - pending
 * - provisioning
 * - scheduled
 */
export const cancelDeployment = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/deployments/{id}/cancel',
    ...options
});
/**
 * Declare stage progress
 */
export const declareStageProgress = (options) => (options.client ?? client).patch({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/provisioning/{deployment_id}/status/{stage}/{attempt}',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Create an attempt for a stage
 */
export const createStageAttempt = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/provisioning/{deployment_id}/status/{stage}/{attempt}',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Declare step progress
 */
export const declareStepProgress = (options) => (options.client ?? client).patch({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/provisioning/{deployment_id}/status/{stage}/{attempt}/{step}',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * List Persistent Volume events
 */
export const listPersistentVolumeEvents = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/volume_events',
    ...options
});
/**
 * List all PersistentVolumes
 */
export const listPersistentVolumes = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/volumes',
    ...options
});
/**
 * Create a PersistentVolume
 */
export const createPersistentVolume = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/volumes',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Delete a PersistentVolume
 */
export const deletePersistentVolume = (options) => (options.client ?? client).delete({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/volumes/{id}',
    ...options
});
/**
 * Get a PersistentVolume
 */
export const getPersistentVolume = (options) => (options.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/volumes/{id}',
    ...options
});
/**
 * Update a PersistentVolume
 */
export const updatePersistentVolume = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/volumes/{id}',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * List Domains
 */
export const listDomains = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/domains',
    ...options
});
/**
 * Create Domain
 */
export const createDomain = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/domains',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Delete Domain
 */
export const deleteDomain = (options) => (options.client ?? client).delete({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/domains/{id}',
    ...options
});
/**
 * Get Domain
 */
export const getDomain = (options) => (options.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/domains/{id}',
    ...options
});
/**
 * Update Domain
 */
export const updateDomain = (options) => (options.client ?? client).patch({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/domains/{id}',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Refresh Domain Status
 */
export const refreshDomainStatus = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/domains/{id}/refresh',
    ...options
});
/**
 * List Instance events
 */
export const listInstanceEvents = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/instance_events',
    ...options
});
/**
 * List Instances
 */
export const listInstances = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/instances',
    ...options
});
/**
 * Get Instance
 */
export const getInstance = (options) => (options.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/instances/{id}',
    ...options
});
/**
 * Create Archive
 *
 * Create a signed URL to upload an archive.
 */
export const createArchive = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/archives',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * List App events
 */
export const listAppEvents = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/app_events',
    ...options
});
/**
 * List App
 */
export const listApps = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/apps',
    ...options
});
/**
 * Create App
 */
export const createApp = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/apps',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Delete App
 *
 * App deletion is allowed for all status.
 */
export const deleteApp = (options) => (options.client ?? client).delete({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/apps/{id}',
    ...options
});
/**
 * Get App
 */
export const getApp = (options) => (options.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/apps/{id}',
    ...options
});
/**
 * Update App
 */
export const updateApp2 = (options) => (options.client ?? client).patch({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/apps/{id}',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Update App
 */
export const updateApp = (options) => (options.client ?? client).put({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/apps/{id}',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Pause App
 *
 * App pause action is allowed for the following status:
 * - starting
 * - healthy
 * - degraded
 * - unhealthy
 * - resuming
 */
export const pauseApp = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/apps/{id}/pause',
    ...options
});
/**
 * Resume App
 *
 * App resume action is allowed for the following status:
 * - paused
 */
export const resumeApp = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/apps/{id}/resume',
    ...options
});
/**
 * DEPRECATED: Review Organization Capacity
 */
export const reviewOrganizationCapacity = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/quotas/capacity',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * List Service events
 */
export const listServiceEvents = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/service_events',
    ...options
});
/**
 * List Services
 */
export const listServices = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/services',
    ...options
});
/**
 * Create Service
 */
export const createService = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/services',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Autocomplete definition
 *
 * Generate autocomplete definition for a service
 */
export const autocomplete = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/services-autocomplete',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Delete Service
 *
 * Service deletion is allowed for all status.
 */
export const deleteService = (options) => (options.client ?? client).delete({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/services/{id}',
    ...options
});
/**
 * Get Service
 */
export const getService = (options) => (options.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/services/{id}',
    ...options
});
/**
 * Update Service
 */
export const updateService2 = (options) => (options.client ?? client).patch({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/services/{id}',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Update Service
 */
export const updateService = (options) => (options.client ?? client).put({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/services/{id}',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Pause Service
 *
 * Service pause action is allowed for the following status:
 * - starting
 * - healthy
 * - degraded
 * - unhealthy
 * - resuming
 */
export const pauseService = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/services/{id}/pause',
    ...options
});
/**
 * ReDeploy Service
 */
export const reDeploy = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/services/{id}/redeploy',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Resume Service
 *
 * Service resume action is allowed for the following status:
 * - paused
 */
export const resumeService = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/services/{id}/resume',
    ...options
});
/**
 * List Secrets
 */
export const listSecrets = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/secrets',
    ...options
});
/**
 * Create Secret
 */
export const createSecret = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/secrets',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Delete Secret
 */
export const deleteSecret = (options) => (options.client ?? client).delete({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/secrets/{id}',
    ...options
});
/**
 * Get Secret
 */
export const getSecret = (options) => (options.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/secrets/{id}',
    ...options
});
/**
 * Update Secret
 */
export const updateSecret2 = (options) => (options.client ?? client).patch({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/secrets/{id}',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Update Secret
 */
export const updateSecret = (options) => (options.client ?? client).put({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/secrets/{id}',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Reveal Secret
 */
export const revealSecret = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/secrets/{id}/reveal',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * List all Snapshots
 */
export const listSnapshots = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/snapshots',
    ...options
});
/**
 * Create a Snapshot
 */
export const createSnapshot = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/snapshots',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Delete a Snapshot
 */
export const deleteSnapshot = (options) => (options.client ?? client).delete({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/snapshots/{id}',
    ...options
});
/**
 * Get a Snapshot
 */
export const getSnapshot = (options) => (options.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/snapshots/{id}',
    ...options
});
/**
 * Update a Snapshot
 */
export const updateSnapshot = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/snapshots/{id}',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Search
 */
export const search = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/search',
    ...options
});
/**
 * List Activities
 */
export const listActivities = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/activities',
    ...options
});
/**
 * List Notifications
 */
export const listNotifications = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/notifications',
    ...options
});
/**
 * List Region
 */
export const listRegions = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/catalog/regions',
    ...options
});
/**
 * Get Region
 */
export const getRegion = (options) => (options.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/catalog/regions/{id}',
    ...options
});
/**
 * List Instances
 */
export const listCatalogInstances = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/catalog/instances',
    ...options
});
/**
 * Get Instance
 */
export const getCatalogInstance = (options) => (options.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/catalog/instances/{id}',
    ...options
});
/**
 * List datacenters
 */
export const listDatacenters = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/catalog/datacenters',
    ...options
});
export const listUsage = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/catalog/usage',
    ...options
});
/**
 * Exec Command
 *
 * This endpoint opens a websocket. Once open, all frames going through the websocket should be formatted in JSON. Input frames should match the format specified below. Output frames will match the response schema.
 */
export const execCommand = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/streams/instances/exec',
    ...options
});
/**
 * Get Metrics
 */
export const getMetrics = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/streams/metrics',
    ...options
});
/**
 * Query logs
 */
export const queryLogs = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/streams/logs/query',
    ...options
});
/**
 * Tails logs
 */
export const tailLogs = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/streams/logs/tail',
    ...options
});
/**
 * Login user
 */
export const login = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/login',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Logout user
 */
export const logout = (options) => (options?.client ?? client).delete({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/logout',
    ...options
});
/**
 * Refresh token
 */
export const refreshToken = (options) => (options?.client ?? client).put({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/refresh',
    ...options
});
/**
 * New session
 *
 * Creates a new session without an organization for current user.
 * NOTE: If you want a session linked to another organization, please use "Switch organization".
 */
export const newSession = (options) => (options?.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/session',
    ...options
});
/**
 * Get intercom profile
 */
export const getIntercomProfile = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/intercom/profile',
    ...options
});
/**
 * Delete user
 */
export const deleteUser = (options) => (options.client ?? client).delete({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/users/{id}',
    ...options
});
/**
 * V2 Delete user
 */
export const deleteUserV2 = (options) => (options.client ?? client).delete({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v2/users/{id}',
    ...options
});
/**
 * List payment methods
 */
export const listPaymentMethods = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/payment_methods',
    ...options
});
/**
 * Create payment authorization
 */
export const createPaymentAuthorization = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/payment_methods',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Delete payment method
 */
export const deletePaymentMethod = (options) => (options.client ?? client).delete({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/payment_methods/{id}',
    ...options
});
/**
 * Get payment method
 */
export const getPaymentMethod = (options) => (options.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/payment_methods/{id}',
    ...options
});
/**
 * Confirm payment authorization
 */
export const confirmPaymentAuthorization = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/payment_methods/{id}/confirm',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Redeem Coupon
 *
 * This API allows to redeem a coupon. Pass the code you received in the body.
 */
export const redeemCoupon = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/coupons',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * List credentials
 */
export const listCredentials = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/credentials',
    ...options
});
/**
 * Create credential
 */
export const createCredential = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/credentials',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Delete credential
 */
export const deleteCredential = (options) => (options.client ?? client).delete({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/credentials/{id}',
    ...options
});
/**
 * Get credential
 */
export const getCredential = (options) => (options.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/credentials/{id}',
    ...options
});
/**
 * Update credential
 */
export const updateCredential2 = (options) => (options.client ?? client).patch({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/credentials/{id}',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Update credential
 */
export const updateCredential = (options) => (options.client ?? client).put({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/credentials/{id}',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Get Subscription
 */
export const getSubscription = (options) => (options.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/subscriptions/{id}',
    ...options
});
/**
 * List Organization Invitations
 */
export const listOrganizationInvitations = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organization_invitations',
    ...options
});
/**
 * Create Organization Invitation
 */
export const createOrganizationInvitation = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organization_invitations',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Delete Organization Invitation
 */
export const deleteOrganizationInvitation = (options) => (options.client ?? client).delete({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organization_invitations/{id}',
    ...options
});
/**
 * Get Organization Invitation
 */
export const getOrganizationInvitation = (options) => (options.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organization_invitations/{id}',
    ...options
});
/**
 * Resend Organization Invitation
 */
export const resendOrganizationInvitation = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organization_invitations/{id}/resend',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Confirm organization action
 */
export const confirmOrganizationAction = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organization_confirmations/{id}',
    ...options
});
export const getQuotas = (options) => (options.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organizations/{organization_id}/quotas',
    ...options
});
/**
 * Get organization usage summary
 */
export const getOrganizationSummary = (options) => (options.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organizations/{organization_id}/summary',
    ...options
});
/**
 * List organization members
 */
export const listOrganizationMembers = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organization_members',
    ...options
});
/**
 * Remove an organization member
 */
export const removeOrganizationMember = (options) => (options.client ?? client).delete({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organization_members/{id}',
    ...options
});
export const getAccountActivities = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/activities',
    ...options
});
/**
 * Begin a session with iDenfy, emit an authToken
 */
export const getIdenfyToken = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/idenfy',
    ...options
});
/**
 * ClearIdenfyVerificationResult marks the current result for idenfy as superseded
 */
export const clearIdenfyVerificationResult = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/idenfy',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
export const createInvite = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/invite',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Get the login method for an email address
 */
export const loginMethod = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/login_method',
    ...options
});
/**
 * Get OAuth Providers
 */
export const getOAuthOptions = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/oauth',
    ...options
});
/**
 * Authenticate using OAuth
 */
export const oAuthCallback = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/oauth',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Get Current Organization
 */
export const getCurrentOrganization = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/organization',
    ...options
});
/**
 * List User Organization Invitations
 */
export const listUserOrganizationInvitations = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/organization_invitations',
    ...options
});
/**
 * Get User Organization Invitation
 */
export const getUserOrganizationInvitation = (options) => (options.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/organization_invitations/{id}',
    ...options
});
/**
 * Accept Organization Invitation
 */
export const acceptOrganizationInvitation = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/organization_invitations/{id}/accept',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Decline Organization Invitation
 */
export const declineOrganizationInvitation = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/organization_invitations/{id}/decline',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * List User Organizations
 *
 * List all organizations that the current user is a member of.
 */
export const listUserOrganizations = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/organizations',
    ...options
});
/**
 * Get Current User
 */
export const getCurrentUser = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/profile',
    ...options
});
/**
 * Update User
 */
export const updateUser2 = (options) => (options.client ?? client).patch({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/profile',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Update User
 */
export const updateUser = (options) => (options.client ?? client).put({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/profile',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Resend Email Verification
 */
export const resendEmailValidation = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/resend_validation',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Reset Password
 */
export const resetPassword = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/reset_password',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
export const getUserSettings = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/settings',
    ...options
});
export const updateUserSettings = (options) => (options.client ?? client).patch({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/settings',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Signup
 */
export const signup = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/signup',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Update Password
 */
export const updatePassword = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/update_password',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Validate
 */
export const validate = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/account/validate/{id}',
    ...options
});
/**
 * Experimental: Has unpaid invoices
 *
 * WARNING: Please don't use the following method.
 * Koyeb doesn't guarantee backwards compatible breaking change and reserve the right to completely drop it without notice.
 * USE AT YOUR OWN RISK.
 */
export const hasUnpaidInvoices = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/billing/has_unpaid_invoices',
    ...options
});
export const manage = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/billing/manage',
    ...options
});
/**
 * Experimental: Fetch next invoice
 *
 * WARNING: Please don't use the following method.
 * Koyeb doesn't guarantee backwards compatible breaking change and reserve the right to completely drop it without notice.
 * USE AT YOUR OWN RISK.
 */
export const nextInvoice = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/billing/next_invoice',
    ...options
});
/**
 * Fetch Github Installation configuration
 */
export const getGithubInstallation = (options) => (options?.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/github/installation',
    ...options
});
/**
 * Start Github Installation
 */
export const githubInstallation = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/github/installation',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Create Organization
 */
export const createOrganization = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organizations',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Delete an Organization
 */
export const deleteOrganization = (options) => (options.client ?? client).delete({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organizations/{id}',
    ...options
});
/**
 * Get Organization
 */
export const getOrganization = (options) => (options.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organizations/{id}',
    ...options
});
/**
 * Update Organization
 */
export const updateOrganization2 = (options) => (options.client ?? client).patch({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organizations/{id}',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Update Organization
 */
export const updateOrganization = (options) => (options.client ?? client).put({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organizations/{id}',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * CreateAccessToken
 *
 * CreateAccessToken creates a short-lived access token in the scope of the
 * specified organization, provided the user making the request is part of
 * said organization.
 *
 * It's possible to specify a validity for the token, which defaults to 1h
 * and must be no more than 24h. The format is `<number>s`, where `<number>`
 * is a floating point in seconds (so `123.456789012s` means 123 seconds and
 * 456789012 nanoseconds). See: https://protobuf.dev/reference/php/api-docs/Google/Protobuf/Duration.html.
 */
export const createAccessToken = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organizations/{id}/access_token',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Deactivate an Organization
 */
export const deactivateOrganization = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organizations/{id}/deactivate',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Update Organization plan
 */
export const updateOrganizationPlan = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organizations/{id}/plan',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Reactivate an Organization
 */
export const reactivateOrganization = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organizations/{id}/reactivate',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Upsert Organization's signup qualification
 */
export const upsertSignupQualification = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organizations/{id}/signup_qualification',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Switch Organization context
 */
export const switchOrganization = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organizations/{id}/switch',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Delete Budget
 */
export const deleteBudget = (options) => (options.client ?? client).delete({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organizations/{organization_id}/budget',
    ...options
});
/**
 * Get Budget
 */
export const getBudget = (options) => (options.client ?? client).get({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organizations/{organization_id}/budget',
    ...options
});
/**
 * Create Budget
 */
export const createBudget = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organizations/{organization_id}/budget',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Update Budget
 */
export const updateBudget = (options) => (options.client ?? client).put({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/organizations/{organization_id}/budget',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
export const cannyAuth = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/sso/canny',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
export const discourseAuth = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/sso/discourse',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Unscope Organization Token
 *
 * UnscopeOrganizationToken removes the organization scope from a token. This
 * endpoint is useful when a user wants to remove an organization: by
 * unscoping the token first, the user can then delete the organization
 * without invalidating his token.
 */
export const unscopeOrganizationToken = (options) => (options.client ?? client).post({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v1/unscope_organization_token',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Update User V2
 */
export const updateUserV22 = (options) => (options.client ?? client).patch({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v2/account/profile',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
/**
 * Update User V2
 */
export const updateUserV2 = (options) => (options.client ?? client).put({
    security: [{ name: 'Authorization', type: 'apiKey' }],
    url: '/v2/account/profile',
    ...options,
    headers: {
        'Content-Type': '*/*',
        ...options.headers
    }
});
