// @koyeb/sandbox-sdk/lib/api.js
import * as koyeb from '@koyeb/api-client-js';
import { DEFAULT_API_HOST } from './constants.js';
import { formatRequest, formatResponse } from './format.js';
import { assert, getEnv } from './utils.js';
export class KoyebApi {
    token;
    debug;
    baseUrl;
    constructor(token, debug = process.env.KOYEB_DEBUG === 'true') {
        this.token = token;
        this.debug = debug;
        this.baseUrl = getEnv('KOYEB_API_HOST') ?? DEFAULT_API_HOST;
    }
    async api(promise) {
        const result = await promise;
        if ('error' in result) {
            throw result.error;
        }
        return result.data;
    }
    get params() {
        return {
            baseUrl: this.baseUrl,
            auth: `Bearer ${this.token}`,
            fetch: this.fetch,
        };
    }
    fetch = async (request) => {
        assert(request instanceof Request);
        if (this.debug) {
            console.debug(await formatRequest(request));
        }
        const response = await globalThis.fetch(request);
        if (this.debug) {
            console.debug(await formatResponse(response));
            console.debug();
        }
        return response;
    };
    async getApp(id) {
        const response = await this.api(koyeb.getApp({ ...this.params, path: { id } }));
        return response.app;
    }
    async createApp(body) {
        const response = await this.api(koyeb.createApp({ ...this.params, body }));
        return response.app;
    }
    async deleteApp(id) {
        const response = await this.api(koyeb.deleteApp({ ...this.params, path: { id } }));
        return response.app;
    }
    async listServices(query) {
        const response = await this.api(koyeb.listServices({ ...this.params, query }));
        return response.services;
    }
    async getService(id) {
        const response = await this.api(koyeb.getService({ ...this.params, path: { id } }));
        return response.service;
    }
    async createService(body, query) {
        const response = await this.api(koyeb.createService({ ...this.params, query, body }));
        return response.service;
    }
    async updateService(id, body, query) {
        const response = await this.api(koyeb.updateService({ ...this.params, path: { id }, query, body }));
        return response.service;
    }
    async deleteService(id) {
        const response = await this.api(koyeb.deleteService({ ...this.params, path: { id } }));
        return response.service;
    }
    async getDeployment(id) {
        const response = await this.api(koyeb.getDeployment({ ...this.params, path: { id } }));
        return response.deployment;
    }
}
