// @koyeb/sandbox-sdk/lib/errors.js
import { PORT_MAX, PORT_MIN } from './constants.js';
export class MissingApiTokenError extends Error {
    constructor() {
        super('API token is required. Set KOYEB_API_TOKEN environment variable or pass api_token parameter');
    }
}
export class InvalidPortError extends Error {
    constructor(value) {
        super(`Port must be an integer between ${PORT_MIN} and ${PORT_MAX}, got ${value}`);
    }
}
export class SandboxTimeoutError extends Error {
    name;
    timeout;
    constructor(name, timeout) {
        super([
            `Sandbox '${name}' did not become ready within ${timeout} seconds.`,
            'The sandbox was created but may not be ready yet.',
            'You can check its status with sandbox.is_healthy() or call sandbox.wait_ready() again.',
        ].join(' '));
        this.name = name;
        this.timeout = timeout;
    }
}
export class NoSandboxSecretError extends Error {
    constructor() {
        super('The SANDBOX_SECRET environment variable is not set');
    }
}
export class SandboxRequestError extends Error {
    response;
    body;
    constructor(response, body) {
        super(`Sandbox executor request failed: ${response.status} ${response.statusText}`);
        this.response = response;
        this.body = body;
    }
}
