// @koyeb/sandbox-sdk/lib/sandbox.js
import { KoyebApi } from './api.js';
import { DEFAULT_IDLE_TIMEOUT, DEFAULT_POLL_INTERVAL, DEFAULT_WAIT_TIMEOUT, PORT_MAX, PORT_MIN } from './constants.js';
import { InvalidPortError, MissingApiTokenError, NoSandboxSecretError, SandboxRequestError, SandboxTimeoutError, } from './errors.js';
import { SandboxFilesystem } from './sandbox-filesystem.js';
import { handleServerSentEvents } from './server-sent-event.js';
import { assert, getEnv, isDefined, isUndefined, omitUndefined, parseDuration, randomString, waitFor, } from './utils.js';
export class Sandbox {
    app_id;
    service_id;
    name;
    sandbox_secret;
    api_token;
    api;
    domain;
    constructor(app_id, service_id, name, sandbox_secret, api_token) {
        this.app_id = app_id;
        this.service_id = service_id;
        this.name = name;
        this.sandbox_secret = sandbox_secret;
        this.api_token = api_token;
        this.api = new KoyebApi(this.api_token);
    }
    get id() {
        return this.service_id;
    }
    static defaultCreateSandboxOptions = {
        image: 'koyeb/sandbox',
        name: 'quick-sandbox',
        wait_ready: true,
        instance_type: 'micro',
        exposed_port_protocol: 'http',
        timeout: DEFAULT_WAIT_TIMEOUT,
        idle_timeout: DEFAULT_IDLE_TIMEOUT,
    };
    static async create(options = {}) {
        const opts = { ...this.defaultCreateSandboxOptions, ...omitUndefined(options) };
        const token = opts.api_token ?? getEnv('KOYEB_API_TOKEN');
        if (!token) {
            throw new MissingApiTokenError();
        }
        const definition = {
            name: opts.name,
            type: 'SANDBOX',
            docker: {
                image: opts.image,
                privileged: opts.privileged,
                image_registry_secret: opts.registry_secret,
            },
            instance_types: [{ type: opts.instance_type }],
            regions: [opts.region ?? 'na'],
            ports: [
                { port: 3030, protocol: 'http' },
                { port: 3031, protocol: opts.exposed_port_protocol },
            ],
            routes: [
                { port: 3030, path: '/koyeb-sandbox/' },
                { port: 3031, path: '/' },
            ],
        };
        const sandbox_secret = randomString(32);
        definition.env ??= [];
        definition.env.push({ key: 'SANDBOX_SECRET', value: sandbox_secret });
        for (const [key, value] of Object.entries(opts.env ?? {})) {
            definition.env.push({ key, value });
        }
        if (opts.idle_timeout > 0) {
            let sleep_idle_delay;
            if (opts._experimental_enable_light_sleep) {
                sleep_idle_delay = { light_sleep_value: opts.idle_timeout, deep_sleep_value: 3900 };
            }
            else {
                sleep_idle_delay = { deep_sleep_value: opts.idle_timeout };
            }
            definition.scalings = [{ min: 0, max: 1, targets: [{ sleep_idle_delay }] }];
        }
        else {
            definition.scalings = [{ min: 1, max: 1 }];
        }
        if (opts.enable_tcp_proxy) {
            definition.proxy_ports = [{ port: 3031, protocol: 'tcp' }];
        }
        const service = await this.createService(token, opts, definition);
        const sandbox = new Sandbox(service.app_id, service.id, service.name, sandbox_secret, token);
        if (opts.wait_ready && !(await sandbox.wait_ready(opts.timeout))) {
            throw new SandboxTimeoutError(sandbox.name, opts.timeout);
        }
        return sandbox;
    }
    static async createService(token, opts, definition) {
        const api = new KoyebApi(token);
        await api.createService({ app_id: '74140198-4d29-4a1e-bdc9-5cc2b355ccd0', definition }, { dry_run: true });
        const app = await api.createApp({
            name: `sandbox-app-${opts.name}-${Date.now()}`,
            life_cycle: {},
        });
        try {
            return await api.createService({
                app_id: app.id,
                definition,
                life_cycle: {
                    delete_after_create: parseDuration(opts.delete_after_delay),
                    delete_after_sleep: parseDuration(opts.delete_after_inactivity_delay),
                },
            });
        }
        catch (error) {
            await api.deleteApp(app.id);
            throw error;
        }
    }
    static async get_from_id(serviceId, apiToken) {
        const token = apiToken ?? getEnv('KOYEB_API_TOKEN');
        if (!token) {
            throw new MissingApiTokenError();
        }
        const api = new KoyebApi(token);
        const service = await api.getService(serviceId);
        const deployment = await api.getDeployment(service.latest_deployment_id);
        const secret = deployment.definition?.env?.find(({ key }) => key === 'SANDBOX_SECRET');
        assert(secret?.value, new NoSandboxSecretError());
        return new Sandbox(service.app_id, service.id, service.name, secret.value, token);
    }
    async wait_ready(timeout = DEFAULT_WAIT_TIMEOUT, pollInterval = DEFAULT_POLL_INTERVAL, signal) {
        return waitFor(() => this.is_healthy(), timeout, pollInterval, signal);
    }
    async wait_tcp_proxy_ready(timeout = DEFAULT_WAIT_TIMEOUT, pollInterval = DEFAULT_POLL_INTERVAL, signal) {
        return waitFor(async () => isDefined(await this.get_tcp_proxy_info()), timeout, pollInterval, signal);
    }
    async is_healthy() {
        const url = await this.get_sandbox_url();
        const response = await fetch(`${url}/health`);
        return response.ok;
    }
    async get_tcp_proxy_info() {
        const service = await this.api.getService(this.service_id);
        if (isUndefined(service.active_deployment_id)) {
            return;
        }
        const deployment = await this.api.getDeployment(service.active_deployment_id);
        const proxy_port = deployment.metadata?.proxy_ports?.find(({ port }) => port === 3031);
        if (!proxy_port) {
            return;
        }
        return [proxy_port.host, proxy_port.public_port];
    }
    async get_domain() {
        if (this.domain) {
            return this.domain;
        }
        const app = await this.api.getApp(this.app_id);
        const domain = app.domains?.[0];
        assert(domain);
        this.domain = domain;
        return this.domain;
    }
    async get_sandbox_url() {
        const domain = await this.get_domain();
        const name = domain.name;
        assert(name);
        return `https://${name}/koyeb-sandbox`;
    }
    async update_lifecycle(values) {
        const service = await this.api.getService(this.service_id);
        const deployment = await this.api.getDeployment(service.latest_deployment_id);
        await this.api.updateService(this.service_id, {
            definition: deployment.definition,
            life_cycle: {
                delete_after_create: parseDuration(values?.delete_after_delay),
                delete_after_sleep: parseDuration(values?.delete_after_inactivity_delay),
            },
        });
    }
    async delete() {
        await this.api.deleteApp(this.app_id);
    }
    async fetch(path, init, requestBody) {
        init.headers = new Headers(init.headers);
        init.headers.set('Authorization', `Bearer ${this.sandbox_secret}`);
        if (isDefined(requestBody)) {
            init.headers.set('Content-Type', 'application/json');
            init.body = JSON.stringify(requestBody);
        }
        return fetch(`${await this.get_sandbox_url()}${path}`, init);
    }
    async request(path, init, requestBody) {
        const response = await this.fetch(path, init, requestBody);
        const contentType = response.headers.get('Content-Type');
        const responseBody = contentType?.startsWith('application/json') ? await response.json() : await response.text();
        if (!response.ok) {
            throw new SandboxRequestError(response, responseBody);
        }
        return responseBody;
    }
    get filesystem() {
        return new SandboxFilesystem(this);
    }
    async exec(cmd, { cwd, env, signal } = {}) {
        return this.request('/run', { method: 'POST', signal }, { cmd, cwd, env });
    }
    exec_stream(cmd, { cwd, env, signal } = {}) {
        const emitter = new EventTarget();
        this.fetch('/run_streaming', { method: 'POST', signal }, { cmd, cwd, env })
            .then((response) => response.body)
            .then((body) => body && handleServerSentEvents(emitter, body))
            .catch((error) => emitter.dispatchEvent(new MessageEvent('error', { data: error })));
        return emitter;
    }
    async expose_port(port) {
        assert(port >= PORT_MIN && port <= PORT_MAX, new InvalidPortError(port));
        await this.unexpose_port();
        await this.request('/bind_port', { method: 'POST' }, { port: String(port) });
        const domain = await this.get_domain();
        assert(isDefined(domain.name));
        return {
            port,
            exposed_at: `https://${domain.name}`,
        };
    }
    async unexpose_port(port) {
        if (isDefined(port)) {
            assert(port >= PORT_MIN && port <= PORT_MAX, new InvalidPortError(port));
        }
        await this.request('/unbind_port', { method: 'POST' }, { port: isDefined(port) ? String(port) : undefined });
    }
    async launch_process(cmd, options) {
        const response = await this.request('/start_process', { method: 'POST' }, { cmd, ...options });
        return response.id;
    }
    async kill_process(processId) {
        await this.request(`/kill_process`, { method: 'POST' }, { id: processId });
    }
    async list_processes() {
        const response = await this.request('/list_processes', { method: 'GET' });
        return response.processes;
    }
    async kill_all_processes() {
        let count = 0;
        for (const process of await this.list_processes()) {
            if (process.status === 'running') {
                await this.kill_process(process.id);
                count++;
            }
        }
        return count;
    }
}
