// @koyeb/sandbox-sdk/lib/sandbox-filesystem.js
export class SandboxFilesystem {
    sandbox;
    constructor(sandbox) {
        this.sandbox = sandbox;
    }
    async mkdir(path, recursive = false) {
        await this.sandbox.request('/make_dir', { method: 'POST' }, { path, recursive });
    }
    async list_dir(path = '.') {
        const result = await this.sandbox.request('/list_dir', { method: 'POST' }, { path });
        return result.entries;
    }
    async delete_dir(path) {
        await this.sandbox.request('/delete_dir', { method: 'POST' }, { path });
    }
    async write_file(path, content) {
        await this.sandbox.request('/write_file', { method: 'POST' }, { path, content });
    }
    async write_files(files) {
        await Promise.all(files.map(({ path, content }) => this.write_file(path, content)));
    }
    async read_file(path) {
        return this.sandbox.request('/read_file', { method: 'POST' }, { path });
    }
    async rename_file(old_path, new_path) {
        await this.sandbox.exec(`mv ${old_path} ${new_path}`);
    }
    async rm(path, recursive = false) {
        if (recursive) {
            await this.sandbox.exec(`rm -rf ${path}`);
        }
        else {
            await this.sandbox.exec(`rm ${path}`);
        }
    }
    async exists(path) {
        const result = await this.sandbox.exec(`test -e ${path}`);
        return result.code === 0;
    }
    async is_file(path) {
        const result = await this.sandbox.exec(`test -f ${path}`);
        return result.code === 0;
    }
    async is_dir(path) {
        const result = await this.sandbox.exec(`test -d ${path}`);
        return result.code === 0;
    }
    async upload_file(local_path, remote_path) {
        const fs = await import('node:fs/promises');
        const content = await fs.readFile(local_path);
        await this.write_file(remote_path, content.toString());
    }
    async download_file(local_path, remote_path) {
        const fs = await import('node:fs/promises');
        const fileInfo = await this.read_file(remote_path);
        await fs.writeFile(local_path, fileInfo.content);
    }
}
