// @koyeb/sandbox-sdk/lib/utils.js
export function isUndefined(value) {
    return value === undefined;
}
export function isDefined(value) {
    return value !== undefined;
}
export function assert(condition, error = new Error('Assertion failed')) {
    if (!condition) {
        throw error;
    }
}
export function omitUndefined(object) {
    return Object.fromEntries(Object.entries(object).filter(([_, value]) => value !== undefined));
}
export function createArray(length, init) {
    return Array(length)
        .fill(null)
        .map((_, index) => init(index));
}
export function randomFloat(max) {
    return Math.random() * max;
}
export function randomInteger(max) {
    return Math.floor(randomFloat(max));
}
export function randomItem(items) {
    return items[randomInteger(items.length - 1)];
}
export function wait(ms, signal) {
    return new Promise((resolve) => {
        const timeout = setTimeout(() => resolve(true), ms);
        if (signal) {
            signal.addEventListener('abort', () => {
                clearTimeout(timeout);
                resolve(false);
            });
        }
    });
}
export async function waitFor(predicate, timeout, interval, signal) {
    const start = Date.now();
    do {
        if (await predicate()) {
            return true;
        }
        await wait(interval, signal);
    } while (Date.now() - start < timeout);
    return false;
}
export function getEnv(name) {
    if (isDefined(typeof process)) {
        return process.env[name];
    }
}
export function nanoId(alphabet) {
    const letters = alphabet.split('');
    return (length) => {
        return createArray(length, () => randomItem(letters)).join('');
    };
}
export const randomString = nanoId('-_0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz');
export function parseDuration(input) {
    if (input === undefined || typeof input === 'number') {
        return input;
    }
    const match = /^(\d+)(s|m|h|d)?$/.exec(input.trim());
    if (!match) {
        throw new Error(`Invalid duration: ${input}`);
    }
    const value = Number(match[1]);
    const unit = match[2];
    return {
        s: value,
        m: value * 60,
        h: value * 60 * 60,
        d: value * 60 * 60 * 24,
    }[unit];
}
